test/test_compiler.py

```diff
@@ -46,11 +45,49 @@ def test_multi_table_delete_multiple(self):
         )
 
     def test_multi_table_update(self):
-        statement = table1.update()\
-                    .values(name=table2.c.name)\
-                    .where(table1.c.id == table2.c.name)
+        statement = table1.update() \
+            .values(name=table2.c.name) \
+            .where(table1.c.id == table2.c.name)
         self.assert_compile(
             statement,
             "UPDATE table1 SET name=table2.name FROM table2 "
             "WHERE table1.id = table2.name"
         )
+
+
+def test_quoted_name_label(engine_testaccount):
+    # quote label
+    col = column('colname').label(quoted_name('alias', True))
+    sel_from_tbl = select([col]).group_by(col).select_from(
+        table(quoted_name('abc', True)))
+    compiled_result = sel_from_tbl.compile(engine_testaccount)
+
+    assert str(compiled_result) == """SELECT colname AS "alias" 
+FROM "abc" GROUP BY colname"""
+
+    # Not quote label
+    col = column('colname').label('alias')
+    sel_from_tbl = select([col]).group_by(col).select_from(
+        table(quoted_name('abc', True)))
+    compiled_result = sel_from_tbl.compile(engine_testaccount)
+
+    assert str(compiled_result) == """SELECT colname AS alias 
+FROM "abc" GROUP BY colname"""
+
+    # Not quote label in capital case
+    col = column('colname').label('ALIAS')
+    sel_from_tbl = select([col]).group_by(col).select_from(
+        table(quoted_name('abc', True)))
+    compiled_result = sel_from_tbl.compile(engine_testaccount)
+
+    assert str(compiled_result) == """SELECT colname AS "ALIAS" 
+FROM "abc" GROUP BY colname"""
+
```

Comment:
I believe having a mixed-case test case might be better, as all uppercase in this case would in fact not impact the results (do correct me if I am wrong). So perhaps the following might be more important to test. Also here remove quoting from table name:

```python
    # Not quote label in mixed case
    col = column('colname').label('Alias')
    sel_from_tbl = select([col]).group_by(col).select_from('abc')
    compiled_result = sel_from_tbl.compile(engine_testaccount)
     assert str(compiled_result) == """SELECT colname AS "Alias" 
FROM abc GROUP BY colname"""
```

Reply:
makes sense.